### ISSUE TITLE
Removing Minimum Qty Allowed in Shopping Cart options triggers PHP error on Change Status Action

### ISSUE THREAD

**Issue by rk-mxp** [magento/magento2#22927]
### Preconditions (*)

1. Magento 2.3.x
2. No Option for ALL GROUPS is set for configuration "Minimum Qty Allowed in Shopping Cart"

### Steps to reproduce (*)

1. Remove all options from Minimum Qty Allowed in Shopping Cart in Configuration - Catalog - Inventory - Product Stock Options
2. Add any product
3. Select product and choose the Update Attributes Action

### Expected result (*)

1. Attributes for the products can be changed.

### Actual result (*)

1. PHP error: Undefined offset: 32000 in .../vendor/magento/module-catalog/view/adminhtml/templates/catalog/product/edit/action/inventory.phtml on line 37

The problem ist the following line:
```
$defaultMinSaleQty = (float) $defaultMinSaleQty[\Magento\Customer\Api\Data\GroupInterface::CUST_GROUP_ALL] ?? 1;
```

The cast to float takes precedence over the Null Coalesce Operator, therefore triggering the error. Change the line to

```
$defaultMinSaleQty = (float) ($defaultMinSaleQty[\Magento\Customer\Api\Data\GroupInterface::CUST_GROUP_ALL] ?? 1);
```

Adding parentheses is required to give the Null Coalesce Operator precedence over the float cast.

Maybe related to issue [21643](https://github.com/magento/magento2/pull/21643).

**Comment by shikhamis11** [comment]
hello @Barakur 
thanks for your report this issue is already confirmed you can check 
#17592 #21640

**Comment by Larke12** [comment]
This task was closed and marked as a duplicate but is only related in scope. The issue here even with the fixes from those issues still stands if a site does not have a default minimum quantity configuration set, which is still the case on both the 2.3 and 2.3-develop branches. The parenthesis noted in the initial comment of this issue will resolve the undefined offset error, while [#17592 ](https://github.com/magento/magento2/issues/17592) resolves the non-numeric errors. This fix should still be applied to prevent errors in this case. The fix for posterity once again:

`$defaultMinSaleQty = (float) ($defaultMinSaleQty[\Magento\Customer\Api\Data\GroupInterface::CUST_GROUP_ALL] ?? 1);`

**Comment by micwallace** [comment]
I can confirm this patch works as expected. The code is obviously semantically incorrect and doesn't really need verification. Please include patch in the next release.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-21867` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @Barakur. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24944` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
